src/utils/messages.ts

```diff
@@ -12,13 +12,17 @@ export function groupMessagesByShortSpanTime(
   messages: EveryMessage[]
 ): EveryMessage[] {
   // Create an object to group messages based on their creation time
-  const groupedMessagesByCreatedAt = messages.reduce((groups, message) => {
-    const { createdAt } = message;
+  const groupedMessagesByCreatedAt = messages.reduce((groups, message, idx) => {
+    const { createdAt, sender } = message;
     // Get the key of the previous group
     const prevKey = Object.keys(groups)[Object.keys(groups).length - 1];
 
     // Check if the time difference between the current message and the previous one is within 3 minutes
-    if (prevKey && message.createdAt - Number(prevKey) <= TIME_SPAN) {
+    if (
+      prevKey &&
+      createdAt - Number(prevKey) <= TIME_SPAN &&
+      sender?.userId === messages[idx - 1]?.sender.userId
+    ) {
```

Comment:
Can admin messages be sent through the AI widget as well?
In UIKit grouping, there is filtering for admin messages, so I'm just checking to see if it's possible.

Reply:
For example

```
// GOOD
(profile)Bot
                         Admin
(profile)Bot
                                        User(-------)
                                        User(profile)
(-------)Bot
(profile)Bot
```

```
// BAD
(-------)Bot
                         Admin
(profile)Bot
                                        User(-------)
                                        User(profile)
(-------)Bot
(profile)Bot
```

Reply:
Oh thanks Airen! https://github.com/sendbird/chat-ai-widget/pull/55/commits/7fab34ba1214ecad6d05206b2b1522f89d5aaa62 just updated the condition to filter out the admin message.

Reply:
++ if that's the case, we should add optional chaining to sender

```diff
- messages[idx - 1]?.sender.userId
+ messages[idx - 1]?.sender?.userId
```